src/glue/jobs/s3_to_json.py

```diff
@@ -0,0 +1,209 @@
+
+    Metadata fields derived from the file basename are inserted as top-level fields
+    and the NDJSON is written to a JSON dataset in S3.
+
+    Args:
+        z (zipfile.Zipfile): The zip archive as provided by the data provider.
+        json_path (str): A JSON path relative to the root of `z`.
+        dataset_identifier (str): The data type of `json_path`.
+        metadata (dict): Metadata derived from the file basename.
+        workflow_run_properties (dict): The workflow arguments
+
+    Returns:
+        output_path (str) The local path the file was written to.
+    """
+    s3_client = boto3.client("s3")
+    logger = logging.getLogger(__name__)
+    os.makedirs(dataset_identifier, exist_ok=True)
+    s3_metadata = metadata.copy()
+    if s3_metadata["start_date"] is None:
+        s3_metadata.pop("start_date")
+    else:
+        s3_metadata["start_date"] = metadata["start_date"].isoformat()
+    s3_metadata["end_date"] = metadata["end_date"].isoformat()
+    data = []
+    with z.open(json_path, "r") as p:
+        for json_line in p:
+            j = json.loads(json_line)
+            j["export_start_date"] = s3_metadata.get("start_date")
+            j["export_end_date"] = s3_metadata.get("end_date")
+            if dataset_identifier == "HealthKitV2Samples":
+                # This puts the `Type` property back where Apple intended it to be
+                j["Type"] = metadata["subtype"]
+            data.append(j)
+    if dataset_identifier == "HealthKitV2Samples":
+        output_fname = "{}_{}_{}-{}.ndjson".format(
+                metadata["type"],
+                metadata["subtype"],
+                metadata["start_date"].strftime("%Y%m%d"),
+                metadata["end_date"].strftime("%Y%m%d")
+        )
+    elif metadata["start_date"] is None:
+        output_fname = "{}_{}.ndjson".format(
+                metadata["type"],
+                metadata["end_date"].strftime("%Y%m%d")
+    )
+    else:
+        output_fname = "{}_{}-{}.ndjson".format(
+                metadata["type"],
+                metadata["start_date"].strftime("%Y%m%d"),
+                metadata["end_date"].strftime("%Y%m%d")
+        )
+    output_path = os.path.join(dataset_identifier, output_fname)
+    s3_output_key = os.path.join(
+        workflow_run_properties["json_prefix"],
+        f"dataset={dataset_identifier}",
+        output_fname)
+    logger.debug(f"Output Key: {s3_output_key}")
+    with open(output_path, "w") as f_out:
+        for record in data:
+            f_out.write("{}\n".format(json.dumps(record)))
+    with open(output_path, "rb") as f_in:
+        response = s3_client.put_object(
+                Body = f_in,
+                Bucket = workflow_run_properties["json_bucket"],
+                Key = s3_output_key,
+                Metadata = s3_metadata)
+        logger.debug(f"S3 Put object response: {json.dumps(response)}")
+    return output_path
+
+def get_metadata(basename):
```

Comment:
```suggestion
def get_metadata(basename: str) -> dict:
```

Reply:
When you add the type hints, vscode has a module that can auto generate the docstrings actually.  I would look into that.